### PR TITLE
feat: Implement CLIENT UNBLOCK

### PR DESCRIPTION
**Pull request by siyopao**
Part of #174.

### CODE REVIEW

**Comment by siyopao** [comment]
Hmm, `acl_test` failed. It looks like the CI is now using Redis 6.2 which is affecting some of the existing tests.

**Comment by siyopao** [comment]
Okay, created https://github.com/denodrivers/redis/pull/196. If that looks good I can rebase on it after.

File: tests/connection_test.ts [hunk 1]
```diff
@@ -123,4 +123,37 @@ suite.test("client tracking", async () => {
...
+suite.test("client unblock with error", async () => {
+  const tempClient = await newClient({ hostname: "127.0.0.1", port: 7003 });
+  try {
+    const id = await tempClient.clientID();
+    assertThrowsAsync(
+      () => tempClient.brpop(0, "key1"),
+      Error,
+      "-UNBLOCKED",
+    );
+    await new Promise((r) => setTimeout(r, 100)); // give some leeway for the command to reach redis
```
Comment (uki00a) [inline tests/connection_test.ts:152]: Ditto

**Review by uki00a — CHANGES_REQUESTED** [review]
Looks good! Just a minor nitpick:

File: tests/connection_test.ts [hunk 2]
```diff
@@ -123,4 +123,37 @@ suite.test("client tracking", async () => {
...
+  const id = await client.clientID();
+  assertEquals(await client.clientUnblock(id), 0);
+});
+
+suite.test("client unblock with timeout", async () => {
+  const tempClient = await newClient({ hostname: "127.0.0.1", port: 7003 });
+  try {
+    const id = await tempClient.clientID();
+    tempClient.brpop(0, "key1"); // block
+    await new Promise((r) => setTimeout(r, 100)); // give some leeway for the command to reach redis
```
[the remark this answers is not in the document]
Comment (siyopao) [reply tests/connection_test.ts]: That is much nicer. Thanks. Fixed!

File: tests/connection_test.ts (hunk 1 above)
Comment (siyopao) [reply tests/connection_test.ts]: Thanks. Fixed here as well.

**Review by uki00a — APPROVED** [review]
LGTM, thanks @siyopao! :+1:

**Pull request merged**